lib/dialects/oracle/utils.js

```diff
@@ -1,25 +1,42 @@
-function generateCombinedName(logger, postfix, name, subNames) {
-  const crypto = require('crypto');
-  const limit = 30;
-  if (!Array.isArray(subNames)) subNames = subNames ? [subNames] : [];
-  const table = name.replace(/\.|-/g, '_');
-  const subNamesPart = subNames.join('_');
-  let result = `${table}_${
-    subNamesPart.length ? subNamesPart + '_' : ''
-  }${postfix}`.toLowerCase();
-  if (result.length > limit) {
-    logger.warn(
-      `Automatically generated name "${result}" exceeds ${limit} character ` +
-        `limit for Oracle. Using base64 encoded sha1 of that name instead.`
-    );
-    // generates the sha1 of the name and encode it with base64
-    result = crypto
-      .createHash('sha1')
-      .update(result)
-      .digest('base64')
-      .replace('=', '');
+class NameHelper {
+  constructor(oracleVersion) {
+    this.oracleVersion = oracleVersion;
+
+    const versionParts = oracleVersion
+      .split('.')
+      .map((versionPart) => parseInt(versionPart));
+    if (
```

Comment:
comment explaining the difference would be helpful

Reply:
Let me know if the comment I added is in line with what you were thinking @kibertoad